tests/test_dimensions.py

```diff
@@ -18,7 +18,40 @@ def test_dimensions():
             dims.TIME: -2.0,
         }
     )
-    assert d3.dimensions == {dims.MASS: 1.0, dims.LENGTH: -1.0, dims.TIME: -2.0}
+    assert d3._dimensions == {dims.MASS: 1.0, dims.LENGTH: -1.0, dims.TIME: -2.0}
+
+
+def test_add():
+    temp = ansunits.Dimensions(
+        dimensions_container={ansunits.BaseDimensions.TEMPERATURE: 1.0}
+    )
+    delta_temp = ansunits.Dimensions(
+        dimensions_container={ansunits.BaseDimensions.TEMPERATURE_DIFFERENCE: 1.0}
+    )
+    not_temp = ansunits.Dimensions(
+        dimensions_container={ansunits.BaseDimensions.MASS: 1.0}
+    )
+
+    assert temp + delta_temp == temp
+    assert temp + temp == None
+    assert temp + not_temp == None
```

Comment:
The correct behaviour is to throw here. If we return `None`, thhen we end up with some obscure exception, separated from the source problem, later on. I you would like to retain this temporarily, you can use my `assert_wrongly` function which is a tracking device. It's **currently** under lib_compare.util.

Reply:
There is no `delta_temp + delta_temp` test (I see it's `None`).

Reply:
Should I move these tests to the Quantity tests? Then all the errors will be raised as expected.

Reply:
I am happy to test it all levels. This is the essence of unit testing. If we decide that `Dimensions` behaviour is wrong then we should fix it at that level, and with tests in place.

Reply:
But you don't need to fix it in this PR